src/compass/utils/cloud_optimized_geotiff.py

```diff
@@ -0,0 +1,77 @@
+import os
+import shutil
+import tempfile
+import journal
+from osgeo import gdal, osr
+from compass.utils.validate_cloud_optimized_geotiff import main as validate_cog
+
+def save_as_cog(filename, scratch_dir = '.', flag_compress=True,
+                resamp_algorithm=None):
+    """Save (overwrite) a GeoTIFF file as a cloud-optimized GeoTIFF.
+       Parameters
+       ----------
+       filename: str
+              GeoTIFF to be saved as a cloud-optimized GeoTIFF
+       scratch_dir: str (optional)
+              Temporary Directory
+       flag_compress: bool (optional)
+              Flag to indicate whether images should be
+              compressed
+       resamp_algorithm: str (optional)
+              Resampling algorithm. Options: "AVERAGE",
+              "AVERAGE_MAGPHASE", "RMS", "BILINEAR",
+              "CUBIC", "CUBICSPLINE", "GAUSS", "LANCZOS",
+              "MODE", "NEAREST", or "NONE"
+    """
+    warning_channel = journal.warning('cloud_optimized_geotiff.save_as_cog')
+    info_channel = journal.info('cloud_optimized_geotiff.save_as_cog')
+
+    info_channel.log('COG step 1: add overviews')
+    gdal_ds = gdal.Open(filename, 1)
+    gdal_dtype = gdal_ds.GetRasterBand(1).DataType
+    dtype_name = gdal.GetDataTypeName(gdal_dtype).lower()
+    is_integer = 'byte' in dtype_name  or 'int' in dtype_name
+
+    overviews_list = [4, 16, 64, 128]
+
+    if is_integer:
+        resamp_algorithm = 'NEAREST'
+
+    gdal_ds.BuildOverviews(resamp_algorithm, overviews_list,
+                           gdal.TermProgress_nocb)
+
+    del gdal_ds  # close the dataset (Python object and pointers)
+    external_overview_file = filename + '.ovr'
+    if os.path.isfile(external_overview_file):
+        os.path.remove(external_overview_file)
+
+    info_channel.log('COG step 2: save as COG')
+    temp_file = tempfile.NamedTemporaryFile(
+                    dir=scratch_dir, suffix='.tif').name
+
+    tile_size = 512
+    ovr_tile_size = tile_size
+    gdal_translate_options = [
+        'TILED=YES',
+        f'BLOCKXSIZE={tile_size}',
+        f'BLOCKYSIZE={tile_size}',
+        f'GDAL_TIFF_OVR_BLOCKSIZE={ovr_tile_size}'
```

Comment:
```suggestion
    gdal_translate_options = [
        'TILED=YES',
        f'BLOCKXSIZE={tile_size}',
        f'BLOCKYSIZE={tile_size}',
        f'GDAL_TIFF_OVR_BLOCKSIZE={tile_size}'
```
Clean up: remove `ovr_tile_size` since it can't be anything but `tile_size`.